### PR TITLE
AE-489: UMA Feedback

### PR DESCRIPTION
**Pull request by arbitrarylink**
add unit tests for agent.js
mint call agent: finding protocol should be something like "uma"
deployer agent: finding protocol should be something like "uma"
when checking the condition `if(!whitelist[to])`, do we need to check the to address of the transaction OR the to address of the trace step?

### CODE REVIEW

File: src/deployer-watch.js [hunk 1]
```diff
@@ -24,8 +24,11 @@ const handleTransaction = async (txEvent) => {
     return findings;
   }
 
+  // low severity alert if the Deployer was involved
+  // txAddresses includes to and from
+  // we are using txAddresses instead of explicitly checking against to and from
+  // to also catch scenarios where deployer is involved but not the intiator
```
Comment (echoAE) [inline src/deployer-watch.js:30]:
```suggestion
  // to also catch scenarios where Deployer is involved but not the initiator
```

**Review by echoAE — COMMENTED** [review]
Small edit in a comment.

**Review by arbitrary-cr0 — COMMENTED** [review]
Tiny tweaks to the comment in deployer-watch

File: src/deployer-watch.js (hunk 1 above)
Comment (arbitrarylink) [reply src/deployer-watch.js]: Done

File: src/deployer-watch.js [hunk 2]
```diff
@@ -24,8 +24,11 @@ const handleTransaction = async (txEvent) => {
     return findings;
   }
 
+  // low severity alert if the Deployer was involved
+  // txAddresses includes to and from
```
[the remark this answers is not in the document]
Comment (arbitrarylink) [reply src/deployer-watch.js]: Done

File: src/deployer-watch.js [hunk 3]
```diff
@@ -24,8 +24,11 @@ const handleTransaction = async (txEvent) => {
...
+  // we are using txAddresses instead of explicitly checking against to and from
```
[the remark this answers is not in the document]
Comment (arbitrarylink) [reply src/deployer-watch.js]: Done

**Review by arbitrary-cr0 — APPROVED** [review]

**Review by arbitrary-kdawg — APPROVED** [review]

**Pull request merged**